### DISCUSSION TITLE
Typescript with AuthConfig callbacks with authState's generic type

### DISCUSSION THREAD

**Discussion by dolbex** [urql-graphql/urql#2480] [Q&A] [answered]
Apologies as I am fairly new to TypeScript and this may be more of a TypeScript question than an urql one. I certainly do appreciate any guidance here.

I'm trying to declare the callback for authConfig for the `willAuthError` method. I'm destructuring the first parameter and pulling out `authState`. `authState` is a generic or null per `authExchange.d.ts`. How do I define that in my callback? This is as close as I've gotten

![Screen Shot 2022-06-06 at 3 36 57 PM](https://user-images.githubusercontent.com/954095/172234565-b25528fb-b471-4bda-8548-1b3c451dc37a.png)

**Comment by JoviDeCroock** [comment, marked as answer]
That's not how generics work though 😅 

if you do

```ts
authExchange<YourAuthStateType>({
  willAuthError
})
```

Then the generic will be inferred for willAuthError else you will need to manually say `authState?: YourAuthStateType`

**Reply by dolbex** [reply]
Awesome, thanks for the help. However, (and I know this will probably induce eyeroll but I'm just totally lost here) - what exactly does `YourAuthStateType` equal? What might that type equal?

**Reply by JoviDeCroock** [reply]
The type you use as an authState for instance `type AuthState = { token?: string }`

**Reply by dolbex** [reply]
Right, of course - that totally worked.
